packages/gateway/src/services/ArbProof.ts

```diff
@@ -0,0 +1,221 @@
+    PublicClient,
+    HttpTransport,
+    Chain,
+    parseAbiItem,
+    toRlp,
+    encodeAbiParameters,
+    parseAbiParameters
+} from 'viem'
+import rollupAbi from './rollupABI'
+
+export interface ArbProvableBlock {
+    number: number
+    sendRoot: string
+    nodeIndex: string
+    rlpEncodedBlock: string
+}
+
+/**
+ * The proofService class can be used to calculate proofs for a given target and slot on the Arbitrum network.
+ * It's also capable of proofing long types such as mappings or string by using all included slots in the proof.
+ *
+ */
+export class ArbProofService<chain extends Chain>
+    implements IProofService<ArbProvableBlock> {
+    private readonly l2Provider: PublicClient<HttpTransport, chain>
+    private readonly helper: EVMProofHelper<chain>
+    private readonly cache: IBlockCache
+    private readonly l1Provider: PublicClient<HttpTransport, chain>
+    private readonly rollupAddress: string
+
+    constructor(
+        l1Provider: PublicClient<HttpTransport, chain>,
+        l2Provider: PublicClient<HttpTransport, chain>,
+        l2RollupAddress: string,
+        cache: IBlockCache,
+    ) {
+        this.l2Provider = l2Provider
+        this.rollupAddress = l2RollupAddress
+        this.l1Provider = l1Provider
+        this.helper = new EVMProofHelper(l2Provider)
+        this.cache = cache
+    }
+
+    async getStorageAt(
+        block: ArbProvableBlock,
+        address: Address,
+        slot: bigint,
+    ): Promise<Hash> {
+        return this.helper.getStorageAt(
+            block.number as unknown as bigint,
+            address,
+            slot,
+        )
+    }
+
+    /**
+     * @dev Fetches a set of proofs for the requested state slots.
+     * @param block A `ProvableBlock` returned by `getProvableBlock`.
+     * @param address The address of the contract to fetch data from.
+     * @param slots An array of slots to fetch data for.
+     * @returns A proof of the given slots, encoded in a manner that this service's
+     *   corresponding decoding library will understand.
+     */
+    async getProofs(
+        block: ArbProvableBlock,
+        address: Address,
+        slots: bigint[],
+    ): Promise<Hash> {
+        const proof = await this.helper.getProofs(
+            block.number as unknown as bigint,
+            address,
+            slots,
+        )
+
+        return encodeAbiParameters(
+            parseAbiParameters([
+                '(bytes32 version, bytes32 sendRoot, uint64 nodeIndex,bytes rlpEncodedBlock)',
+                '(bytes[] stateTrieWitness, bytes[][] storageProofs)'
+            ]),
+            [
+                {
+                    version:
+                        '0x0000000000000000000000000000000000000000000000000000000000000000',
+                    sendRoot: block.sendRoot,
+                    nodeIndex: block.nodeIndex,
+                    rlpEncodedBlock: block.rlpEncodedBlock,
+                },
+                proof,
+            ] as never
```

Comment:
why this?

Reply:
Currently, the conversion is not need. Good point Lucas.